File: JiangG_Project02/src/JiangG_Project02_Main.java

import java.io.*;
/*
@Name :Gangjun Jiang
@Student ID :23795633
@date 1/31/2023
@file name : JiangG_Project01_Main
* */
class JiangG_Project02_Main {

    private static void processString(BufferedReader inFile2, BufferedWriter outFile2) throws IOException {
        String line;
        int count = 0;
        int total =0;
        outFile2.write("In processInt method:\n");
        outFile2.newLine();

        while ((line = inFile2.readLine()) != null) {
            String[] words = line.split(" ");
            for (int i = 0; i < words.length; i++) {
                outFile2.write(words[i] + " ");
                count++;
                if (count % 5 == 0) {
                    outFile2.newLine();
                    count = 0;
                } total++;
            }
        }
        outFile2.write( "\n the total string count is " + total);
    }
    public static void main(String[] args) {
        try {
// Open the input text file

                File inFile2 = new File(args[0]);
                //File inFile2 = new File("C:\\Users\\jjram\\IdeaProjects\\JiangG_Project02\\src\\data2.txt");
                BufferedReader inFile = new BufferedReader(new FileReader(inFile2));
                // Open the output text file
                File outFile2 = new File(args[1]);
                //File outFile2 = new File("C:\\Users\\jjram\\IdeaProjects\\JiangG_Project02\\src\\outFile2.txt");
                BufferedWriter outFile = new BufferedWriter(new FileWriter(outFile2));


                processString(inFile, outFile);

                inFile.close();
                outFile.close();

        } catch (Exception e) {
            e.printStackTrace();
        }
    }

}

File: JiangG_Project02_C++/JiangG_Project02_Main.cpp
#include <fstream>
#include <iostream>
#include <string>
using namespace std;
/*
@Name :Gangjun Jiang
@Student ID :23795633
@date 1/31/2023
@file name : JiangG_Project02_Main
* */
class JiangG_Project02_Main {
public:
    string data;
    int total, count;
    void processString(ifstream& inFile2, ofstream& outFile2) {
        outFile2 << "In processString method:" << endl;
        total = 0;
        count = 0;
        while (inFile2 >> data) {
            outFile2 << data << " ";
            count++;
            if (count >= 5) {
                outFile2 << endl;
                count = 0;
            }total++;
        }
        outFile2 << "\n the total string count is " << total << endl;
    }
};
int main( int argc, char* argv[]) {
// Open the input text file
    ifstream inFile2;
    ofstream outFile2;
    JiangG_Project02_Main JiangG_Project02_Main;
    inFile2.open(argv[1]);
    outFile2.open(argv[2]);
    JiangG_Project02_Main.processString(inFile2, outFile2);
    inFile2.close();
    outFile2.close();
    return 0;
}